src/progressbar/progressbar.ts

```diff
@@ -0,0 +1,36 @@
+import {Component, Input, ElementRef} from 'angular2/angular2';
+
+@Component({
+  selector: 'ngb-progressbar',
+  template: `
+    <progress class="progress {{extraClasses}}  {{type && 'progress-' + type}}" [value]="value" [max]="max">
+      <div class="progress">
+        <span class="progress-bar" [style.width.%]="calculatePorcentage()"><ng-content></ng-content></span>
+      </div>
+    </progress>
+  `
+})
+export class NgbProgressbar {
+  _value: number;
+  min = 0;
+  extraClasses: string;
+
+  @Input()
+  set value(newValue: number) {
+    this._value = newValue;
+    this.calculatePorcentage();
+  }
+
+  get value() { return this._value; }
+
+  @Input() type: string;
+
+  @Input() max = 100;
+
+  constructor(private element: ElementRef) {
```

Comment:
So what ng-class does is, gets `class` as an input (so you get all the classes on the parent), but to delete it from the parent, it is toggling them. So yes, it is manually deleting the classes on the host element and needed a ElementRef and Renderer.

The idea of a striped attribute is something I had in mind too, but the problem is that it will work for this use case, but this topic will come back for sure. Still, not sure if there is a clean solution (not even in ng1).